### PR TITLE
Add experimental InfluxDB write support.

### PR DESCRIPTION
**Pull request by juliusv**

### CODE REVIEW

**Comment by juliusv** [comment]
@beorn7

File: main.go [hunk 1]
```diff
@@ -50,8 +51,9 @@ var (
 
 	persistenceStoragePath = flag.String("storage.local.path", "/tmp/metrics", "Base path for metrics storage.")
 
-	remoteTSDBUrl     = flag.String("storage.remote.url", "", "The URL of the OpenTSDB instance to send samples to.")
-	remoteTSDBTimeout = flag.Duration("storage.remote.timeout", 30*time.Second, "The timeout to use when sending samples to OpenTSDB.")
+	remoteStorageType    = flag.String("storage.remote.type", "opentsdb", "The type of remote storage to use. Valid values: 'opentsdb', 'influxdb'.")
+	remoteStorageURL     = flag.String("storage.remote.url", "", "The URL of the remote storage instance to send samples to.")
```
Comment (beorn7) [inline main.go]: This assumes that all future storage will have a URL. But others might need other parameters (let's say a mysql-style DSN) or even more than only one parameter (location of a credential file, other config parameters). How about replacing "storage.remote.type" and "storage.remote.url" by "storage.remote.opentsdb-url" and "storage.remote.influxdb-url"? We could later add "storage.remote.hipsterdb-dsn", "storage.remote.hipsterdb-extent-size" and such.

If more than one of these flags is set, we could either bail out or even send samples to all of them... (the latter is simpler for now).

File: storage/remote/influxdb/client.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
+package influxdb
```
Comment (beorn7) [inline storage/remote/influxdb/client.go]: Add copyright note.

File: storage/remote/influxdb/client_test.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
+// Copyright 2013 The Prometheus Authors
```
Comment (beorn7) [inline storage/remote/influxdb/client_test.go]: s/2013/2015/

**Comment by beorn7** [comment]
Otherwise :+1:  (modulo "I don't know nothing about influxdb.")

File: main.go (hunk 1 above)
Comment (juliusv) [inline main.go]: Sounds good. How about then moving the decision whether to do long-term storage from checking whether one of these URLs is set to checking whether `storage.remote.type` is empty, and setting that to an empty string by default?

File: storage/remote/influxdb/client_test.go (hunk 3 above)
Comment (juliusv) [inline storage/remote/influxdb/client_test.go]: Done, thanks.

File: storage/remote/influxdb/client.go (hunk 2 above)
Comment (juliusv) [inline storage/remote/influxdb/client.go]: Done, and also added in the `opentsdb/client.go` file.

File: main.go (hunk 1 above)
Comment (juliusv) [inline main.go]: Done.

**Comment by juliusv** [comment]
PTAL.

File: main.go [hunk 4]
```diff
@@ -50,8 +51,10 @@ var (
 
 	persistenceStoragePath = flag.String("storage.local.path", "/tmp/metrics", "Base path for metrics storage.")
 
-	remoteTSDBUrl     = flag.String("storage.remote.url", "", "The URL of the OpenTSDB instance to send samples to.")
-	remoteTSDBTimeout = flag.Duration("storage.remote.timeout", 30*time.Second, "The timeout to use when sending samples to OpenTSDB.")
+	remoteStorageType    = flag.String("storage.remote.type", "", "The type of remote storage to use. Valid values: 'opentsdb', 'influxdb'. If this flag is left empty, no remote storage is used.")
```
Comment (beorn7) [inline main.go]: My idea was to get rid of this flag completely.

A remote storage is activated by providing its flag(s) below.

If flags for more than one are created, we either send to each of them, or we bail out with an error.

Comment (matthiasr) [inline main.go]: sending to all of them is what I would expect, and would be extremely helpful e.g. in migration scenarios.

**Comment by juliusv** [comment]
Updated to allow using OpenTSDB and InfluxDB at the same time.

**Comment by beorn7** [comment]
:+1:

**Pull request merged**